content/doc/book/pipeline/jenkinsfile.adoc

```diff
@@ -889,6 +853,175 @@ within a single `withCredentials( ... ) { ... }` step by doing the following:
   `withCredentials( ... ) { ... }` step snippet.
 
 
+=== String interpolation
+
+Jenkins Pipeline uses rules identical to link:http://groovy-lang.org[Groovy] for
+string interpolation. Groovy's String interpolation support can be confusing to
+many newcomers to the language. While Groovy supports declaring a string with
+either single quotes, or double quotes, for example:
+
+[source,groovy]
+----
+def singlyQuoted = 'Hello'
+def doublyQuoted = "World"
+----
+
+Only the latter string will support the dollar-sign (`$`) based string
+interpolation, for example:
+
+[source,groovy]
+----
+def username = 'Jenkins'
+echo 'Hello Mr. ${username}'
+echo "I said, Hello Mr. ${username}"
+----
+
+Would result in:
+
+[source]
+----
+Hello Mr. ${username}
+I said, Hello Mr. Jenkins
+----
+
+Understanding how to use string interpolation is vital for using some of
+Pipeline's more advanced features.
+
+[WARNING]
```

Comment:
this looks quite weird having such a large warning section, I scroll and scroll and scroll and the page is on the far right of my monitor

I think this needs breaking up and using the warning ⚠️ UI elements more appropriately